### DISCUSSION TITLE
WifI CF-940AX & OpenNept4une (Neptune 4Pro)?

### DISCUSSION THREAD

**Discussion by Xruss** [OpenNeptune3D/OpenNept4une#191] [Q&A]
Good afternoon

I have wi-fi CF-940AX. There are drivers for Linux aic8800fdrvpackage_arm64.deb. How can I install and connect wifi to OpenNept4une (Neptune 4Pro)?

**Comment by halfmanbear** [comment]
Your adapter uses the AIC8800FC chipset
You have the correct deb file

You'll have to copy that deb file to the printer.
Say to the MKS home directory `~/`

Then run;
```
cd ~
sudo apt update
sudo dpkg -i aic8800fdrvpackage_arm64.deb
sudo apt install -f
sudo reboot
```

**Reply by Xruss** [reply]
WI-FI is not detected.
Please tell me how to solve the error?
The driver is taken from
https://en.comfast.com.cn/uploadfile/2023/1220/20231220091945935.zip

mks@znp-k1:~$ cd ~

mks@znp-k1:~$ sudo apt update
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://security.debian.org bookworm-security InRelease
Hit:3 http://deb.debian.org/debian bookworm-updates InRelease
Hit:5 http://deb.debian.org/debian bookworm-backports InRelease
Hit:4 http://fi.mirror.armbian.de/apt bookworm InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
7 packages can be upgraded. Run 'apt list --upgradable' to see them.

mks@znp-k1:~$ sudo dpkg -i aic8800fdrvpackage_arm64.deb
(Reading database ... 88881 files and directories currently installed.)
Preparing to unpack aic8800fdrvpackage_arm64.deb ...
Uninstall aic8800 wifi driver!!!!!
rmmod: ERROR: Module aic8800_fdrv is not currently loaded
rmmod: ERROR: Module aic_load_fw is not currently loaded
/bin/sh: 1: test: -lt: unexpected operator
rm -rf *.o *.ko *.o.* *.mod.* modules.* Module.* .a* .o* .*.o.* *.mod .tmp* .cache.mk
/bin/sh: 1: test: -lt: unexpected operator
rm -rfv /lib/modules/6.1.67-current-rockchip64/kernel/drivers/net/wireless/aic8800/aic8800_fdrv.ko
/sbin/depmod -a 6.1.67-current-rockchip64
rm -rf *.o *.ko *.o.* *.mod.* modules.* Module.* .a* .o* .*.o.* *.mod .tmp* .cache.mk
rm -rfv /lib/modules/6.1.67-current-rockchip64/kernel/drivers/net/wireless/aic8800/aic_load_fw.ko
/sbin/depmod -a 6.1.67-current-rockchip64
rm *.o wifi_test bt_test
Install aic8800 wifi driver!!!!!
Unpacking aic8800fdrvpackage (1.0.1) over (1.0.1) ...
Uninstall aic8800 wifi driver successful!!!!!
Setting up aic8800fdrvpackage (1.0.1) ...
udev done
device exist
/var/lib/dpkg/info/aic8800fdrvpackage.postinst: line 10: eject: command not found
cp fw done
make -C /lib/modules/6.1.67-current-rockchip64/build M=/AIC8800/drivers/aic8800 ARCH=arm64 CROSS_COMPILE= modules
make[1]: *** /lib/modules/6.1.67-current-rockchip64/build: No such file or directory.  Stop.
make: *** [Makefile:55: modules] Error 2
mkdir -p /lib/modules/6.1.67-current-rockchip64/kernel/drivers/net/wireless/aic8800
install -p -m 644 aic_load_fw/aic_load_fw.ko  /lib/modules/6.1.67-current-rockchip64/kernel/drivers/net/wireless/aic8800/
install: cannot stat 'aic_load_fw/aic_load_fw.ko': No such file or directory
make: *** [Makefile:59: install] Error 1
insmod: ERROR: could not load module /AIC8800/drivers/aic8800/aic_load_fw/aic_load_fw.ko: No such file or directory
insmod: ERROR: could not load module /AIC8800/drivers/aic8800/aic8800_fdrv/aic8800_fdrv.ko: No such file or directory
insmod done
gcc -c wifi_test.c -o wifi_test.o
gcc wifi_test.o -o wifi_test
gcc -c bt_test.c -o bt_test.o
gcc bt_test.o -lpthread -o bt_test
sudo cp wifi_test /sbin
sudo cp bt_test /sbin
Install aic8800 wifi driver successful!!!!!

mks@znp-k1:~$ sudo apt install -f
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 7 not upgraded.

mks@znp-k1:~$ sudo reboot
Broadcast message from root@znp-k1 on pts/3 (Mon 2024-12-02 14:17:48 +05):
The system will reboot now!

**Comment by Xruss** [comment]
WI-FI is not detected.
Please tell me how to solve the error?

**Comment by 54iceblock** [comment]
I had the same issue. The header wasn't there so I can't build the driver for my wireless adapter.
